docs/site/src/demos/text-fields/RequiredTextField.js

```diff
@@ -5,11 +5,19 @@ import TextField, { TextFieldInput, TextFieldLabel } from 'material-ui/TextField
 
 export default function RequiredTextField() {
   return (
-    <TextField>
-      <TextFieldLabel htmlFor="required" required>
-        Name
-      </TextFieldLabel>
-      <TextFieldInput id="required" />
-    </TextField>
+    <div>
+      <TextField>
+        <TextFieldLabel htmlFor="required" required>
+          First Name
+        </TextFieldLabel>
+        <TextFieldInput id="required" />
+      </TextField>
+      <TextField required error>
```

Comment:
Note that proper rendering of label and underline in a required + error condition is required at the top level.